modules/iterator/claws/iterator/iterator_view.hpp

```diff
@@ -0,0 +1,140 @@
+    it OP## = val;                                                                                                                                             \
+    return *this;                                                                                                                                              \
+  };                                                                                                                                                           \
+                                                                                                                                                               \
+  constexpr iterator_view operator OP(difference_type val)                                                                                                     \
+    const noexcept(noexcept(std::declval<iterator_view<raw_it_type, func_type>>() OP## = std::declval<difference_type>())                                      \
+                   && std::is_nothrow_copy_constructible_v<iterator_view<raw_it_type, func_type>>)                                                             \
+  {                                                                                                                                                            \
+    auto tmp(*this);                                                                                                                                           \
+                                                                                                                                                               \
+    tmp OP## = val;                                                                                                                                            \
+    return tmp;                                                                                                                                                \
+  };
+
+    CONTAINER_VIEW_ITERATOR_OP(-);
+
+    CONTAINER_VIEW_ITERATOR_OP(+);
+
+    constexpr difference_type operator-(iterator_view const &other) const noexcept(noexcept(std::declval<raw_it_type>() - std::declval<difference_type>()))
+    {
+      return it - other.it;
+    };
+
+    constexpr reference operator*() const noexcept(noexcept(func(*it)))
+    {
+      return func(*it);
+    }
+
+    template<class index_type>
+    constexpr reference operator[](index_type index) const noexcept(noexcept(func(*it)))
+    {
+      return func(*(it + index));
+    }
+
+    /// \todo add noexcept specification
+    constexpr decltype(auto) operator-> () const
+    {
+      struct self_pointer_like
+      {
+        value_type data;
+
+        auto operator-> ()
+        {
+          return &data;
+        }
+      };
+      return self_pointer_like{func(*it)};
+    }
+
+#define CONTAINER_VIEW_ITERATOR_BINARY_PREDICATE(OP)                                                                                                           \
```

Comment:
Might wanna update macro name

Reply:
There is another such macro in this file I had missed

Reply:
Yup, I plan to do so in upcoming commits, as this one is only about moving files around and formatting